app/lib/katello/util/support.rb

```diff
@@ -78,12 +78,12 @@ def self.with_db_connection(logger = Rails.logger)
         logger.error(e.message)
         logger.error("Lost database connection. Attempting reconnect.")
 
-        active_record_retry_connect
+        active_record_retry_connect(logger)
 
         retry
       end
 
-      def self.active_record_retry_connect
+      def self.active_record_retry_connect(logger)
```

Comment:
`logger` was undefined in this method. but it appears to still recover correctly